### PR TITLE
feat: adds support for IUM

### PR DESCRIPTION
**Pull request by jpvajda**
### **Overview**
Added support for the `InjectUserMessage` feature to allow text-based interaction with Deepgram's Agent API. This enables developers to trigger agent responses using text input instead of audio.

### **Files Modified**

**✅ Core Implementation:**
1. **`pkg/api/agent/v1/websocket/interfaces/types.go`**
   - Added `InjectUserMessage` struct with `Type` and `Content` fields
   - Follows same pattern as existing `InjectAgentMessage`

2. **`pkg/api/agent/v1/websocket/interfaces/constants.go`**
   - Added `TypeInjectUserMessage = "InjectUserMessage"` constant

3. **`pkg/client/agent/v1/websocket/types.go`**
   - Added `InjectUserMessage` type alias for client usage

**✅ Testing & Package Fixes:**
4. **`tests/unit_test/agent_test.go`** *(NEW FILE)*
   - Comprehensive test suite with 6 test functions:
     - Struct creation & validation
     - JSON marshaling/unmarshaling
     - Type alias functionality
     - Constants verification  
     - Edge cases (empty content, special characters)

5. **`tests/unit_test/prerecorded_test.go`**
   - Fixed package declaration: `package deepgram_test` → `package deepgram`

6. **`tests/unit_test/mocks.go`**
   - Fixed package declaration: `package deepgram_test` → `package deepgram`

7. Updates Readme with the correct way to run tests in Go.

### **Feature Usage**
```go
// Create and send InjectUserMessage
msg := msginterfaces.InjectUserMessage{
    Type:    msginterfaces.TypeInjectUserMessage,
    Content: "Hello, how can you help me today?",
}
err := wsClient.WriteJSON(msg)
```

### **Design Decisions**
- **Consistency**: Follows exact same pattern as existing `InjectAgentMessage`
- **No convenience method**: Users call `WriteJSON()` directly (matches current SDK patterns)
- **Full test coverage**: Validates JSON serialization, edge cases, and type safety

### **Verification**
- ✅ All tests pass
- ✅ Follows existing SDK conventions  
- ✅ No breaking changes
- ✅ Package naming conflicts resolved

## Types of changes

What types of changes does your code introduce to the community 

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have lint'ed all of my code using repo standards
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: tests/unit_test/agent_test.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
+// Copyright 2024 Deepgram SDK contributors. All Rights Reserved.
```
Comment (lukeocodes) [inline tests/unit_test/agent_test.go:1]: Good tests, you should use this test suite as a guide for the Python SDK

File: tests/unit_test/mocks.go [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a MIT license that can be found in the LICENSE file.
 // SPDX-License-Identifier: MIT
 
-package deepgram_test
```
Comment (lukeocodes) [inline tests/unit_test/mocks.go:5]: uh oh... this will make the tests get published with the deepgram package. i don't think that is right

**Review by lukeocodes — CHANGES_REQUESTED** [review]

File: tests/unit_test/mocks.go (hunk 2 above)
Comment (jpvajda) [reply tests/unit_test/mocks.go:5]: Yeah I had some weird errors with `deepgram_test` so I was playing around and committed this incorrectly. Good catch. Fixed in recent commit.

**Comment by jpvajda** [comment]
I discovered an odd bug in the Agent Implementation which is making unit tests fail using `deepgram_test`


✅ **Works fine with `package deepgram_test`:**
- `simple_test.go` (no special imports)
- `prerecorded_test.go` (imports standard packages)
- `mocks.go` (just mock data)

❌ **Breaks with agent imports:**
- Any file importing `msginterfaces "github.com/deepgram/deepgram-go-sdk/v3/pkg/api/agent/v1/websocket/interfaces"`

This suggests there's something specific about the **agent websocket interfaces package** causing the issue.

I have a theory: **maybe the issue is with the specific import path or module resolution**. 


**Bingo!** The issue happens with **ANY** agent import, not just the interfaces. This suggests there's something fundamentally wrong with the **agent package itself** or there's a **module/import path issue**.

**Next Step**: Let's check what the agent client package declares.
Read file: pkg/client/agent/init.go

## Root Cause Identified

✅ **Works perfectly**: Non-agent imports (`analyze`, `prerecorded`, etc.)  
❌ **Breaks mysteriously**: ANY agent-related imports (`agent`, `msginterfaces`)

## Answer 

**Yes, there IS something specific about the Agent API that doesn't work with unit tests!**

The issue is in the **agent package dependency chain**:

1. `pkg/client/agent/client.go` imports `msginterfaces` 
2. This creates some kind of circular dependency or module resolution conflict
3. When Go tries to resolve the imports, it somehow misreads the test file's package declaration

## The Real Problem

It's likely one of these issues:
- **Circular dependency** in the agent package structure
- **Import cycle** between agent packages
- **Module resolution conflict** in the agent dependency chain
- **Build constraint or build tag issue** in the agent packages

**Comment by jpvajda** [comment]
The Error when running unit tests persists: 

```Run go test -v -run Test_ ./...
found packages deepgram (agent_inject_user_test.go) and deepgram_test (mocks.go) in /home/runner/work/deepgram-go-sdk/deepgram-go-sdk/tests/unit_test
Error: Process completed with exit code 1.
```
 and I'm stuck on how to resolve this after trying for a few hours.

Here's a recap of what I tried:

### Problem Recap

**The Error**: 
```
found packages deepgram (agent_inject_user_test.go) and deepgram_test (mocks.go)
```

**Solutions Tried**:

• **Cache Clearing** (multiple times):
  - `go clean -cache`
  - `go clean -modcache` 
  - Both together

• **File Investigation**:
  - Verified both files use `package deepgram_test` (they do)
  - Checked for hidden characters/encoding issues
  - Hex dump analysis of package declaration lines

**Current Status**: 
- All 3 files correctly declare `package deepgram_test`
- Go still reports package conflict
- Cache clearing doesn't resolve it
- Tests fail to run in CI/CD

**The Real Problem**: Go is somehow detecting a phantom `package deepgram` declaration in `agent_inject_user_test.go` that doesn't actually exist in the file.
